### ISSUE TITLE
Locus app won't work :(

### ISSUE THREAD

**Issue by comfygee2884** [Freeyourgadget/Gadgetbridge#479]
Heya all.
Firstly, since it seems fitbit are gonna shut down the Pebble servers in 2017 and brick my watch, I've found this awesome Gadgetbridge so thank you very VERY much for letting us bypass the authenticaion bits.

But, I use a superb GPS app on my phone called Locus Pro and it has an android plugin and also an offical one on the Pebble store:https://play.google.com/store/apps/details?id=com.gilisztaturmixgmail.locus_addon_pebble&hl=en_GB

Is there any way in the world to get Gagetbridge to recognise this Pebble app please as I am not technically minded and don't understand how to get Pebble app store apps running with Gadgetbridge.

Thanks alot in advance.
G :)

**Comment by IzzySoft** [comment]
You need to install the companion app (the "Pebble Addon" for Locus) on your Android device, and the corresponding watch app on your Pebble¹. Configure the addon, and whenever you fire up Locus the watch app will be loaded.

That far it works for me. Still, I could't figure how to convince it giving me routing information on the Pebble: I start Locus, the watch app is loaded. I start a navigation, but no changes on the watch. As communication per se works (or the watch app wouldn't be loaded automatically), I'm not sure who's to blame here.

----

¹ To install the watch app, you need to [download the `.pbw` file](https://apps.getpebble.com/en_US/application/5781754a6c21044501000f25?section=watchapps&section=watchapps&dev_settings=true&hardware=basalt&is_browser=true&platform=android) to your Android device, then open it with any file explorer – you should then be asked to have the GB Installer used to open it, confirm that.

**Comment by danielegobbetti** [comment]
I added the "not a bug" tag, if you think that this is indeed a bug please let me know and I'll reevaluate this :-)
Happy to read you like GB!

**Comment by IzzySoft** [comment]
@danielegobbetti I'm not sure on which side the culprit sits here. As the basic communication seems to work (the watch-app starts when Locus starts, and Locus can be started from the watch-app) – but navigation data is not passed, it might well be something "in between": something GB does in a way different from the "official app", or a part of some protocol not covered. Unfortunately, that addon is not open-source, so we cannot investigate ourselves. We might ask the dev to drop-in here, though.

**Comment by IzzySoft** [comment]
@danielegobbetti I'm not 100% sure – but as far as I understood, it should get all those from Locus itself. It should display different things, some of which it can only get from Locus (like the target of a navigation). The only thing I've seen it display up to know was the compass needle – where I wasn't even sure whether it pointed North or to the target ;)

TL;DR: even if it would get the location itself, it doesn't even show the data it's supposed to get from its "master app".

**Comment by danielegobbetti** [comment]
OK, will leave it as not a bug and close this: since the app is closed source there is not much we can hunt from our side.
If you can provide a GB link with errors or warnings please feel free to reopen.
Closing as the question by op has been answered.

**Comment by IzzySoft** [comment]
@danielegobbetti If the dev drops in, he could help on the hunt. "won't work" describes the issue quite well, as apart from being switched-on it doesn't currently serve its purpose. So I suggest we reopen this if Attila drops in and needs some cooperation – as I gave him this link and asked him to join. With the fact in mind that the official app won't be continued, he might be interested in working with GB anyhow :smile_cat:

**Comment by IzzySoft** [comment]
Actually, I have to relativise what I wrote. Just tested with the latest versions, and it shows "turns" as well as "target reached" – just not distance etc. Has to do with the settings of the addon, which are sometimes a bit ambiguous: Just checked them again, and I guess next time I'll see what I want :)

**Comment by ashimokawa** [comment]
@comfygee2884 
Maybe you just forget to enable 3rd party support in Gadgetbridge settings?

**Comment by IzzySoft** [comment]
PS: Just got answer from Attila. He'll look into GB as soon as time permits. Told him that's a good idea, as that's the future – especially with the official Pebble app discontinued :smile_cat: 

As v0.16 now allows sharing watch apps to GB, the Pebble Addon for Locus can use that to push the watch-app. Left Attila a hint on that. Will make things easier.
